### PR TITLE
Newsletter

### PR DESCRIPTION
**Pull request by Tithira-Perera-210460V**
Added newsletter section to blog page, individual newsletter page and a function to map locally-stored data to newsletters

### CODE REVIEW

**Comment by Tithira-Perera-210460V** [comment]
Added new Code. Please review.

File: src/app/news/page.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+export default function Page() {
+  const [data, setData] = useState<DataItem[]>([]);
+  const [newsletter, setNewsletter] = useState<DataItem | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await getNewsletters();
+        console.log("Data:", data);
```
Comment (rajivaPavan) [inline src/app/news/page.tsx:18]: Let's remove these logs

File: src/app/news/page.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    //     currentUrl.searchParams.set("key", newsletter.key);
+
+    // // Update browser history (optional)
+    // window.history.pushState({}, currentUrl.toString());
+
+    // console.log("New URL with query parameter:", currentUrl.toString());
+
+    // // Reload the page again with search parameter "key"
+    // window.location.href = currentUrl.toString();
+    //--------------------------------------------------------------------------
```
Comment (rajivaPavan) [inline src/app/news/page.tsx:64]: The comments seem to be obsolete. If they are, let's remove them as well

File: package.json [hunk 3]
```diff
@@ -16,6 +16,8 @@
     "@types/react": "18.2.17",
     "@types/react-dom": "18.2.7",
     "axios": "^1.5.0",
+    "fs": "^0.0.1-security",
+    "fs-extra": "^11.2.0",
```
Comment (rajivaPavan) [inline package.json:20]: Are these packages needed?

**Review by rajivaPavan — CHANGES_REQUESTED** [review]
Looks good to me. Requested some minor changes.

File: package.json (hunk 3 above)
Comment (Tithira-Perera-210460V) [reply package.json:20]: Ah, no. Not required now.

File: src/app/news/page.tsx (hunk 2 above)
Comment (Tithira-Perera-210460V) [reply src/app/news/page.tsx:64]: You can remove it.

File: src/app/news/page.tsx (hunk 1 above)
Comment (Tithira-Perera-210460V) [reply src/app/news/page.tsx:18]: Must have missed. Yup, remove them.

**Review by rajivaPavan — APPROVED** [review]

**Review by rajivaPavan — CHANGES_REQUESTED** [review]
Let's keep this PR as it is for now and merge it after the first newsletter is officially added to the branch

**Pull request merged**